core/src/main/java/dk/alexandra/fresco/lib/lp/LPSolver.java

```diff
@@ -121,145 +112,127 @@ private boolean checkDimensions(LPTableau tableau, Matrix<Computation<SInt>> upd
         enumeratedVariables.add(BigInteger.valueOf(i));
       }
 
-      LPState initialState = new LPState(
-          BigInteger.ZERO, tableau, updateMatrix, null, pivot,
+      LPState initialState = new LPState(BigInteger.ZERO, tableau, updateMatrix, null, pivot,
           enumeratedVariables, initialBasis, pivot);
       return () -> initialState;
-    }).whileLoop(
-        state -> !state.terminated(),
-        (state, seq) -> {
-          iterations++;
-          if (debugLog) {
-            debugInfo(seq, state);
-          }
-          return seq.seq((inner) -> {
-            logger.info("LP Iterations=" + iterations + " solving " + identityHashCode);
-            if (pivotRule == PivotRule.BLAND) {
-              return blandPhaseOneProtocol(inner, state);
-            } else {
-              return phaseOneProtocol(inner, state, zero);
-            }
-          }).seq((phaseOneOutput, inner) -> {
-            if (!phaseOneOutput.terminated()) {
-              phaseTwoProtocol(inner, phaseOneOutput);
-            }
-            return phaseOneOutput;
-          });
-        }).seq((whileState, seq) ->
-        () -> new LPOutput(whileState.tableau, whileState.updateMatrix, whileState.basis,
-            whileState.pivot)
-    );
+    }).whileLoop(state -> !state.terminated(), (state, seq) -> {
```

Comment:
This is not a good realignment, now the condition is obscured by the lambda construction